packages/babel-generator/src/whitespace.js

```diff
@@ -42,19 +17,11 @@ export default class Whitespace {
     let endToken;
     let tokens = this.tokens;
 
-    for (let j = 0; j < tokens.length; j++) {
-      // optimize for forward traversal by shifting for loop index
-      let i = getLookupIndex(j, this._lastFoundIndex, this.tokens.length);
-      let token = tokens[i];
-
-      // this is the token this node starts with
-      if (node.start === token.start) {
-        startToken = tokens[i - 1];
-        endToken = token;
-
-        this._lastFoundIndex = i;
-        break;
-      }
+    let index = this._findToken(token => token.start - node.start, 0, tokens.length);
+    if (typeof index === "number") {
+      while (index && node.start === tokens[index - 1].start) --index;
```

Comment:
@amasad It will fail if we have a comment at the beginning of the file.

```javascript


// test
// test
```

But we can use `index >= 0` and let `_findToken` returns -1 when token not found, like `[].indexOf`.